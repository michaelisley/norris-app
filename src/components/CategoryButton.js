import React from 'react'

export default function CategoryButton(props) {
    const {handleCategoryButtonClick, category} = props;
    return (
        <button data-testid="button" onClick={() => handleCategoryButtonClick(category)}>{category}</button>
    )
}
