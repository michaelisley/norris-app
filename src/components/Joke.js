import React, { useEffect } from 'react'

export default function Joke(props) {
    const {category, joke, isLoaded, setJoke } = props;
    useEffect(() => {
        fetch(`https://api.chucknorris.io/jokes/random?category=${category}`)
        .then(res => res.json())
        .then(
            (result) => {
                setJoke(category, result.value, true);
            },
            (error) => {
                setJoke(category, error, true);
            }
        )
    }, [])
    if (!isLoaded) {
        return <div className="loading">Loading joke...</div>;
    } else {
        return (
            <>
                <div className="joke">{joke}</div>
                
            </>
        );
    }
}
