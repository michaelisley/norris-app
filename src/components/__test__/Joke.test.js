import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import Joke from './../Joke';
import CategoryButton from './../CategoryButton';

it("renders joke without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Joke></Joke>, div);
})

it("renders button properly", () => {
    const {getByTestId} = render(<CategoryButton category="animal"></CategoryButton>)
    expect(getByTestId('button')).toHaveTextContent("animal");
})

it("matches the snapshot", () => {
    const tree = TestRenderer.create(<CategoryButton category="religion"></CategoryButton>).toJSON();
    expect(tree).toMatchSnapshot();
})