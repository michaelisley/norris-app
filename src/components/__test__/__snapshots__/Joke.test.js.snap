// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<button
  data-testid="button"
  onClick={[Function]}
>
  religion
</button>
`;
